### PR TITLE
gh-102388: Add windows_31j to aliases for cp932 codec

### PR DESCRIPTION
**Pull request by moriyama**
The charset name "Windows-31J" is registered in the IANA Charset Registry[1] and is implemented in Python as the cp932 codec.

This commit adds windows_31j to the aliases of cp932.

[1] https://www.iana.org/assignments/charset-reg/windows-31J

* Issue: gh-102388

### CODE REVIEW

**Review by malemburg — APPROVED** [review]
LGTM.

Perhaps you could fix the many "py:class reference target not found: Codec" errors as part of this PR as well. I'm not really sure what's causing them - not a ResT expert.

File: Doc/library/codecs.rst [hunk 1]
```diff
@@ -1126,7 +1126,8 @@ particular, the following variants typically exist:
 +-----------------+--------------------------------+--------------------------------+
 | cp875           |                                | Greek                          |
 +-----------------+--------------------------------+--------------------------------+
-| cp932           | 932, ms932, mskanji, ms-kanji  | Japanese                       |
+| cp932           | 932, ms932, mskanji, ms-kanji, | Japanese                       |
+|                 | windows_31j                    |                                |
```
Comment (methane) [inline Doc/library/codecs.rst]: In this table, hiphen is used more than underscore. e.g. "iso-2022-jp-2004", "windows-1251".
So "windows-31j" looks more natural to me.

Comment (moriyama) [reply Doc/library/codecs.rst]: Thank you for your advice.
Fixed the description of Doc/library/codecs.rst.

Comment (moriyama) [reply Doc/library/codecs.rst]: The modifications are as follows:
https://github.com/python/cpython/compare/e2371988360b156d2f755e855df06b18cf66f46b..566d1e0403009baa66e47d387c98b073c57f2f0d

**Comment by methane** [comment]
> Perhaps you could fix the many "py:class reference target not found: Codec" errors as part of this PR as well. I'm not really sure what's causing them - not a ResT expert.

There is no `class Codec` in the `Doc/library/codecs.rst`. It should be fixed in other pull request.

**Comment by moriyama** [comment]
Thank you for reviewing and merging.

**Pull request merged**